rm-main/src/ui/mod.rs

```diff
@@ -45,8 +44,9 @@ impl Component for MainWindow {
         use Action as A;
 
         match action {
-            A::Error(e_popup) => {
-                self.global_popup_manager.error_popup = Some(*e_popup);
+            A::Error(error) => {
+                let error_popup = ErrorPopup::new(&error.title, error.message);
+                self.global_popup_manager.error_popup = Some(error_popup);
```

Comment:
possible to use a setter instead?

Reply:
I think another PR (containing several refactors, not just this one) would be better for this since this one is already getting big

Reply:
the PRs at my work get far bigger than this LOL - but sure :))